samples/esb/esb_monitor/src/sniffer.c

```diff
@@ -0,0 +1,159 @@
+{
+	enum esb_sniffer_rtt_comm comm;
+	uint32_t read_len;
+	int err;
+
+	/* Send packet length to backend */
+	read_len = sizeof(struct rtt_frame);
+	read_len = sys_cpu_to_be32(read_len);
+
+	SEGGER_RTT_Write(CONFIG_ESB_SNIFFER_RTT_COMM_RET_CHANNEL,
+						&read_len, sizeof(uint32_t));
+	while (1) {
+		read_len = SEGGER_RTT_Read(CONFIG_ESB_SNIFFER_RTT_COMM_CHANNEL,
+						&comm, sizeof(enum esb_sniffer_rtt_comm));
+		if (read_len == sizeof(enum esb_sniffer_rtt_comm)) {
+			switch (comm) {
+			case SNIFFER_START:
+				LOG_INF("Start sniffer");
+				if (!sniffer_cfg.is_running) {
+					err = esb_start_rx();
+				} else {
+					err = 0;
+				}
+
+				if (err != 0) {
+					LOG_ERR("Failed to start sniffer, err: %d", err);
+				} else {
+					sniffer_cfg.is_running = true;
+				}
+
+				SEGGER_RTT_Write(CONFIG_ESB_SNIFFER_RTT_COMM_RET_CHANNEL,
+										&err, sizeof(int));
+				break;
+			case SNIFFER_STOP:
+				LOG_INF("Stop sniffer");
+				if (sniffer_cfg.is_running) {
+					err = esb_stop_rx();
+				} else {
+					err = 0;
+				}
+
+				if (err != 0) {
+					LOG_ERR("Failed to stop sniffer, err: %d", err);
+				} else {
+					sniffer_cfg.is_running = false;
+				}
+
+				SEGGER_RTT_Write(CONFIG_ESB_SNIFFER_RTT_COMM_RET_CHANNEL,
+										&err, sizeof(int));
+				break;
+			case SNIFFER_STATUS:
+				err = sniffer_cfg.is_running;
+				SEGGER_RTT_Write(CONFIG_ESB_SNIFFER_RTT_COMM_RET_CHANNEL,
+										&err, sizeof(int));
+				break;
+			default:
+				LOG_ERR("Unrecognized rtt command: %d", comm);
+			}
+		}
+
+		k_sleep(K_MSEC(500));
+	}
+}
+
+int sniffer_init(void)
+{
+	int err;
+
+	err = SEGGER_RTT_ConfigUpBuffer(CONFIG_ESB_SNIFFER_RTT_DATA_CHANNEL,
+					"Data", rtt_data_buff,
+					CONFIG_ESB_SNIFFER_RTT_DATA_BUFF_SZ,
+					SEGGER_RTT_MODE_NO_BLOCK_SKIP);
+
+	err += SEGGER_RTT_ConfigUpBuffer(CONFIG_ESB_SNIFFER_RTT_COMM_RET_CHANNEL,
+					 "Comm_ret", rtt_comm_ret_buff,
+					 CONFIG_ESB_SNIFFER_RTT_COMM_RET_BUFF_SZ,
+					 SEGGER_RTT_MODE_NO_BLOCK_SKIP);
+
+	err += SEGGER_RTT_ConfigDownBuffer(CONFIG_ESB_SNIFFER_RTT_COMM_CHANNEL,
+					   "Commands", rtt_comm_buff,
+					   CONFIG_ESB_SNIFFER_RTT_COMM_BUFF_SZ,
+					   SEGGER_RTT_MODE_NO_BLOCK_SKIP);
+	if (err != 0) {
+		LOG_ERR("RTT initialization failed");
+		return -1;
+	}
+
+	/* Setup sniffer_cfg from already initialized ESB */
+	sniffer_cfg.enabled_pipes = 0xFF;
+	sniffer_cfg.bitrate = (enum esb_bitrate)nrf_radio_mode_get(NRF_RADIO);
+	sniffer_cfg.is_running = false;
+
+	radio_reg_to_array(nrf_radio_base0_get(NRF_RADIO), sniffer_cfg.base_addr0);
+	radio_reg_to_array(nrf_radio_base1_get(NRF_RADIO), sniffer_cfg.base_addr1);
+	radio_reg_to_array(nrf_radio_prefix0_get(NRF_RADIO), sniffer_cfg.pipe_prefix);
+	radio_reg_to_array(nrf_radio_prefix1_get(NRF_RADIO), &sniffer_cfg.pipe_prefix[4]);
+
+	rtt_comm_thread_id = k_thread_create(&rtt_comm_thread,
+				rtt_comm_thread_stack,
+				CONFIG_ESB_SNIFFER_THREAD_STACK_SZ,
+				(k_thread_entry_t) rtt_comm_thread_fn,
```

Comment:
```suggestion
				(k_thread_entry_t)rtt_comm_thread_fn,
```
But I would suggest to reformat the function itself to match the type instead of casting.

Reply:
Reformatting function means adding 3 unused void* arguments to definition, I think that casting is more clean solution.